server.js

```diff
@@ -1,34 +1,64 @@
 import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
+import listEndpoints from 'express-list-endpoints'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+import topMusicData from './data/top-music.json'
+
+//   PORT=8080 npm start
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+// returns only songs with danceability of 70 or higher
+app.get('/danceable', (req, res) => {   
+  const danceabilityList = topMusicData.filter((item) => item.danceability >= 70)
+  res.json(danceabilityList)
+})
+
+// Endpoint to get all songs
+app.get('/songs', (req, res) => {   
+  res.json(topMusicData)
+})
+
+// endpoint to get all songs from a specific genre 
+app.get('/genre/:genre', (req, res) => {
+  const { genre } = req.params 
+  const genreList = topMusicData.filter((item) => item.genre.toLowerCase() === genre.toLowerCase())
+  if (genreList.length < 1) {
+    res.status(404).send(`No genre suitable: ${genre}`)
+  } 
+  res.json(genreList)
+})
 
-// Start defining your routes here
+// Endpoint to get songs from specific artist
+app.get('/artist/', (req, res) => {
+  const { artistName } = req.query 
+  if (artistName) {
+    // eslint-disable-next-line max-len
+    const artistList = topMusicData.filter((item) => item.artistName.toLowerCase().includes(artistName))
+    res.json(artistList)
+  }
+  res.json(topMusicData)
+})
```

Comment:
when you want to use a query to filter your data you don't create an extra endpoint. you should add this logic to the `/songs` endpoint.